paymentsheet/src/main/java/com/stripe/android/common/model/CommonConfiguration.kt

```diff
@@ -41,46 +43,83 @@ internal data class CommonConfiguration(
-                            throw IllegalArgumentException(
-                                "Argument does not look like a CustomerSession client secret. " +
-                                    "See CustomerSession API: https://docs.stripe.com/api/customer_sessions/create"
-                            )
-                        }
-                        is CustomerSessionClientSecretValidator.Result.Valid -> Unit
-                    }
-                }
+        when (result) {
+            is CustomerSessionClientSecretValidator.Result.Error.Empty -> {
+                throw IllegalArgumentException(
+                    "When a CustomerConfiguration is passed to PaymentSheet, " +
+                        "the customerSessionClientSecret cannot be an empty string."
+                )
+            }
+            is CustomerSessionClientSecretValidator.Result.Error.LegacyEphemeralKey -> {
+                throw IllegalArgumentException(
+                    "Argument looks like an Ephemeral Key secret, but expecting a CustomerSession client " +
+                        "secret. See CustomerSession API: https://docs.stripe.com/api/customer_sessions/create"
+                )
             }
+            is CustomerSessionClientSecretValidator.Result.Error.UnknownKey -> {
+                throw IllegalArgumentException(
+                    "Argument does not look like a CustomerSession client secret. " +
+                        "See CustomerSession API: https://docs.stripe.com/api/customer_sessions/create"
+                )
+            }
+            is CustomerSessionClientSecretValidator.Result.Valid -> Unit
+        }
+    }
+
+    @Suppress("ThrowsCount")
+    private fun legacyCustomerEphemeralKeyValidate(
+        customerAccessType: PaymentSheet.CustomerAccessType.LegacyCustomerEphemeralKey
+    ) {
+        if (customerAccessType.ephemeralKeySecret != customer?.ephemeralKeySecret) {
+            throw IllegalArgumentException(
+                "Conflicting ephemeralKeySecrets between CustomerConfiguration " +
+                    "and CustomerConfiguration.customerAccessType"
+            )
+        } else if (customerAccessType.ephemeralKeySecret.isBlank() || customer.ephemeralKeySecret.isBlank()) {
+            throw IllegalArgumentException(
+                "When a CustomerConfiguration is passed to PaymentSheet, " +
+                    "the ephemeralKeySecret cannot be an empty string."
+            )
+        } else if (
+            customerAccessType.ephemeralKeySecret.isEKClientSecretValid().not() ||
+            customer.ephemeralKeySecret.isEKClientSecretValid().not()
+        ) {
+            throw IllegalArgumentException(
+                "`ephemeralKeySecret` format does not match expected client secret formatting"
+            )
         }
     }
+
+    fun validate() {
+        // These are not localized as they are not intended to be displayed to a user.
```

Comment:
it's not clear what this comment is referring to anymore -- we should probably either:
- delete this comment
- specify what "these" are (the exception messages)

Either approach is ok with me